src/app/wallets/wallet-name-available.ts

```diff
@@ -0,0 +1,17 @@
+import { CouldNotFindError } from "@domain/errors"
+import { checkedToWalletName } from "@domain/wallets"
+import { WalletsRepository } from "@services/mongoose"
+
+export const walletNameAvailable = async (
+  walletName: string,
+): Promise<boolean | Error> => {
```

Comment:
For the use-case return types we've been using `Promise<type | ApplicationError>`. This being explicit should make it easier to do an exhaustiveness check if we want to. I will fix and merge.